src/algorithms/reference-value.ts

```diff
@@ -0,0 +1,55 @@
+import { ReferenceResult } from '../types';
+const _ = require('lodash');
+
+export function ReferenceValue(
+  x: number,
+  formulae: Array<{
+    is_reference: boolean; // true if this is a reference value
+    method: string; // the method used to calculate the reference value
+    formula: string; // the formula used to calculate the reference value
+    min: number; // the minimum value of the reference value
+    max: number; // the maximum value of the reference value
+  }>
+): ReferenceResult {
+  // find is_reference = true array element
+  const referenceArray = formulae.filter((formula) => formula.is_reference);
+
+  // find the not reference value
+  const notReferenceArray = formulae.filter(
+    (formula) => formula.is_reference === false
+  );
+
+  let rangeValue = referenceArray.filter((r) => {
```

Comment:
instead of rangeValue, you can call this `formulaeInRange` or `matchingFormulae`